src/index/TextIndexBuilder.cpp

```diff
@@ -50,8 +50,25 @@ void TextIndexBuilder::buildTextIndexFile(
   scoreData_ = {vocab_.getLocaleManager(), textScoringMetric_,
                 bAndKParamForTextScoring_};
 
+  // Build the Mmap vector containing `VocabIndex`es of all relevant literals
+  // if it hasn't been build during RDF index building and option to add
+  // literals was set
+  if (!textIndexIndices_.has_value() && addWordsFromLiterals) {
+    textIndexIndices_ = ad_utility::MmapVector<VocabIndex>(
+        onDiskBase_ + TEXT_INDEX_LITERAL_IDS, ad_utility::CreateTag{});
+    for (VocabIndex index = VocabIndex::make(0); index.get() < vocab_.size();
+         index = index.incremented()) {
+      auto text = vocab_[index];
+      if (!isLiteral(text)) {
+        continue;
+      }
+      textIndexIndices_.value().push_back(index);
+    }
```

Comment:
Add a comment that 1. This is currently adds all literals to the text index (filtering settings are ignored), and 2. That this is currently required for some E2E tests.